dandi/download.py

```diff
@@ -309,13 +312,33 @@ def download_generator(
                     lock=lock,
                 )
 
+            # If exception is raised we might just raise it, or yield
+            # an error record
+            gen = {
+                "raise": _download_generator,
+                "yield": _download_generator_guard(path, _download_generator),
+            }[on_error]
+
             if yield_generator_for_fields:
-                yield {"path": path, yield_generator_for_fields: _download_generator}
+                yield {"path": path, yield_generator_for_fields: gen}
             else:
-                for resp in _download_generator:
+                for resp in gen:
                     yield dict(resp, path=path)
 
 
+def _download_generator_guard(path: str, generator: Iterator[dict]) -> Iterator[dict]:
+    try:
+        for resp in generator:
+            yield dict(resp, path=path)
+    except Exception as exc:
+        lgr.error("Caught while downloading %s", path, exc_info=exc)
+        yield {
+            "path": path,
+            "status": "error",
+            "message": str(exc.__class__.__name__),
```

Comment:
Are you sure you don't want to include the error message via `str(exc)`?

Reply:
I went back on this since then it becomes "too verbose" with each message just repeating the known path and summary duplicating all those 
```
$> dandi download DANDI:000029
PATH                                     SIZE     DONE    DONE% CHECKSUM STATUS    MESSAGE              
dandiset.yaml                                                            skipped   no change            
...64/sub-anm369964_behavior+ecephys.nwb                                 error     File './000029/sub...
...63/sub-anm369963_behavior+ecephys.nwb                                 error     File './000029/sub...
...62/sub-anm369962_behavior+ecephys.nwb                                 error     File './000029/sub...
sub-RAT123/sub-RAT123.nwb                                                error     File './000029/sub...
sample.txt                                                               error     File './000029/sam...
sample.json                                                              error     File './000029/sam...
Summary:                                 0 Bytes  0 Bytes                1 skipped 1 no change          
                                         +20.7 MB 0.00%                  6 error   1 File './000029/s...
                                                                                   1 File './000029/s...
                                                                                   1 File './000029/s...
                                                                                   1 File './000029/s...
                                                                                   1 File './000029/s...
                                                                                   1 File './000029/s...
2022-08-03 11:38:10,873 [    INFO] Logs saved in /home/yoh/.cache/dandi-cli/log/20220803153807Z-1022747.log
(dev3) 1 41191.....................................:Wed 03 Aug 2022 11:38:12 AM EDT:.
lena:/tmp
$> dandi download DANDI:000029
PATH                                             SIZE     DONE    DONE% CHECKSUM STATUS    MESSAGE                                                                          
dandiset.yaml                                                                    skipped   no change                                                                        
sub-anm369964/sub-anm369964_behavior+ecephys.nwb                                 error     File './000029/sub-anm369964/sub-anm369964_behavior+ecephys.nwb' already exists  
sub-anm369963/sub-anm369963_behavior+ecephys.nwb                                 error     File './000029/sub-anm369963/sub-anm369963_behavior+ecephys.nwb' already exists  
sub-anm369962/sub-anm369962_behavior+ecephys.nwb                                 error     File './000029/sub-anm369962/sub-anm369962_behavior+ecephys.nwb' already exists  
sub-RAT123/sub-RAT123.nwb                                                        error     File './000029/sub-RAT123/sub-RAT123.nwb' already exists                         
sample.txt                                                                       error     File './000029/sample.txt' already exists                                        
sample.json                                                                      error     File './000029/sample.json' already exists                                       
Summary:                                         0 Bytes  0 Bytes                1 skipped 1 no change                                                                      
                                                 +20.7 MB 0.00%                  6 error   1 File './000029/sub-anm369964/sub-anm369964_behavior+ecephys.nwb' already exists
                                                                                           1 File './000029/sub-anm369963/sub-anm369963_behavior+ecephys.nwb' already exists
                                                                                           1 File './000029/sub-anm369962/sub-anm369962_behavior+ecephys.nwb' already exists
                                                                                           1 File './000029/sub-RAT123/sub-RAT123.nwb' already exists                       
                                                                                           1 File './000029/sample.txt' already exists                                      
                                                                                           1 File './000029/sample.json' already exists     
```

with only reporting the class we have better depiction with details of exception present in the log:

```
$> dandi download DANDI:000029
PATH                                             SIZE     DONE    DONE% CHECKSUM STATUS    MESSAGE          
dandiset.yaml                                                                    skipped   no change        
sub-anm369964/sub-anm369964_behavior+ecephys.nwb                                 error     FileExistsError  
sub-anm369963/sub-anm369963_behavior+ecephys.nwb                                 error     FileExistsError  
sub-anm369962/sub-anm369962_behavior+ecephys.nwb                                 error     FileExistsError  
sub-RAT123/sub-RAT123.nwb                                                        error     FileExistsError  
sample.txt                                                                       error     FileExistsError  
sample.json                                                                      error     FileExistsError  
Summary:                                         0 Bytes  0 Bytes                1 skipped 1 no change      
                                                 +20.7 MB 0.00%                  6 error   6 FileExistsError
2022-08-03 11:41:08,828 [    INFO] Logs saved in /home/yoh/.cache/dandi-cli/log/20220803154105Z-1024172.log
```
```